### PR TITLE
test: add test for mandatory in builders

### PR DESCRIPTION
**Pull request by sdelamo**
This includes a test, and a fix which swallows exceptions thrown by build, which shows the problem.

See: https://github.com/grails/grails-data-mapping/issues/1300

Affects: https://github.com/micronaut-projects/micronaut-groovy/issues/48

### CODE REVIEW

**Comment by sdelamo** [comment]
Maybe we should not log anything and continue silently

File: grails-datastore-core/src/main/groovy/org/grails/datastore/mapping/config/ConfigurationBuilder.groovy [hunk 1]
```diff
@@ -188,7 +188,11 @@ abstract class ConfigurationBuilder<B, C> {
 
                         def buildMethod = ReflectionUtils.findMethod(newBuilder.getClass(), 'build')
                         if (buildMethod != null) {
-                            method.invoke(builder, buildMethod.invoke(newBuilder))
+                            try {
+                                method.invoke(builder, buildMethod.invoke(newBuilder))
+                            } catch(Throwable e) {
+                                log.error("build method threw exception")
```
Comment (erichelgeson) [inline grails-datastore-core/src/main/groovy/org/grails/datastore/mapping/config/ConfigurationBuilder.groovy:194]: could we add `log.error("build method threw exception", e)` so the user can see the stack trace & message?

**Comment by puneetbehl** [comment]
@sdelamo Please update this PR against the 7.0.x branch.

**Comment by sdelamo** [comment]
@puneetbehl I changed the base.

**Pull request merged**